spanner/row.go

```diff
@@ -311,5 +312,43 @@ func (r *Row) ToStruct(p interface{}) error {
 		&sppb.StructType{Fields: r.fields},
 		&proto3.ListValue{Values: r.vals},
 		p,
+		false,
+	)
+}
+
+// ToStructLenient fetches the columns in a row into the fields of a struct.
+// The rules for mapping a row's columns into a struct's exported fields
+// are:
+//
+//   1. If a field has a `spanner: "column_name"` tag, then decode column
+//      'column_name' into the field. A special case is the `spanner: "-"`
+//      tag, which instructs ToStruct to ignore the field during decoding.
+//
+//   2. Otherwise, if the name of a field matches the name of a column (ignoring case),
+//      decode the column into the field.
+//
+//
+// The fields of the destination struct can be of any type that is acceptable
+// to spanner.Row.Column. No errors will be returned if the row's column is not
+// mapped to a struct's exported field or destination struct has extra exported fields,
+// unmatched struct's exported fields will have default values.
+//
+// Slice and pointer fields will be set to nil if the source column is NULL, and a
+// non-nil value if the column is not NULL. To decode NULL values of other types, use
+// one of the spanner.NullXXX types as the type of the destination field.
+func (r *Row) ToStructLenient(p interface{}) error {
```

Comment:
```suggestion
// If ToStruct returns an error, the contents of p are undefined. Some fields may
// have been successfully populated, while others were not; you should not use any of
// the fields.
func (r *Row) ToStructLenient(p interface{}) error {
```

This warning is still true for this method. Although this method is less likely to return an error, it can still fail if for example the type of one of the fields is not compatible with the datatype of the corresponding column.